docs/bugpattern/LockOnNonEnclosingClassLiteral.md

```diff
@@ -1,18 +1,35 @@
-Lock on the class other than the enclosing class of the code block can
-unintentionally prevent the locked class or other classing using the same lock
-being used properly. A issue can be caused when the locked class being edited or
-deleted. And it can also be exhausting and time consuming for the others who is
-using the same class literal as lock to make sure the synchronized blocks can
-work properly as expected. Hence locking on the class other than the enclosing
-class of the synchronized code block is disencouraged by the error prone.
+Having a lock on a class, other than the enclosing class of the code block, can
+unintentionally prevent the locked class from being used properly. This also
+holds for other classes that are using the same lock.
+An issue can arise when the locked class is being edited or deleted.
+Additionally, when others are using the same class literal as lock, it can be
+exhausting and time-consuming for others to ensure that the `synchronized`
+blocks are working as expected. Hence, locking on a class other than the
+enclosing class of the `synchronized` code block is discouraged by Error Prone.
 Locking on the enclosing class or an instance is a preferred practice.
```

Comment:
I was having a bit too much trouble with these paragraph. What is really the issue was not clear to me. Furthermore, some sentences are a bit obsolete/saying the same thing. I think we can minimize this to:

```suggestion
Having a lock on a class, other than the enclosing class of the code block, can 
unintentionally prevent the locked class from being used properly when other
classes effectively lock on the same resource. This can be time-consuming to
handle from a maintainability perspective to ensure the `synchronized` blocks
are working as expected. Hence, locking on a class other than the enclosing
class of the `synchronized` code block is discouraged by Error Prone. Locking
on the enclosing class or an instance is a preferred practice.
```

Reply:
Nice one, let's do it.